### PR TITLE
[KYUUBI #3688] add rest_api and kyuubi-admin docs for admin delete/list engine operation

### LINKED ISSUE #3688: [Subtask] add rest_api and kyuubi-admin docs for admin delete/list engine operation
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

add doc for admin delete/list api and admin ctl.

### Parent issue

https://github.com/apache/incubator-kyuubi/issues/3554

### Are you willing to submit PR?

- [ ] Yes. I can submit a PR independently to improve.
- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lightning-L**
### _Why are the changes needed?_

close #3688 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image
image
image

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by lightning-L** [comment]
@turboFei

**Review by turboFei — APPROVED** [review]
LGTM, could you attach the screenshots into the patch description

**Comment by lightning-L** [comment]
image
image

**Comment by lightning-L** [comment]
image

**Review by turboFei — APPROVED** [review]

File: docs/client/rest/rest_api.md [hunk 1]
```diff
@@ -349,6 +349,43 @@ Gets the local log lines from this batch.
 | logRowSet | The log lines     | List of sting |
 | rowCount  | The log row count | Int           |
 
+## Admin Resource
+
+### POST /admin/refresh/hadoop_conf
+
+Refresh the Kyuubi server hadoop conf, and return the success message.
```
Comment (Yikf) [inline docs/client/rest/rest_api.md:356]: Can simplifies to `Refresh the Kyuubi server hadoop conf` ?

**Review by Yikf — APPROVED** [review]

File: docs/client/rest/rest_api.md [hunk 2]
```diff
@@ -349,6 +349,43 @@ Gets the local log lines from this batch.
 | logRowSet | The log lines     | List of sting |
 | rowCount  | The log row count | Int           |
 
+## Admin Resource
+
+### POST /admin/refresh/hadoop_conf
+
+Refresh the Kyuubi server hadoop conf.
```
Comment (yaooqinn) [inline docs/client/rest/rest_api.md:356]: Refresh the Hadoop configurations of the Kyuubi server.

File: docs/client/rest/rest_api.md [hunk 3]
```diff
@@ -349,6 +349,43 @@ Gets the local log lines from this batch.
...
+
+### DELETE /admin/engine
+
+Delete specified engine.
```
Comment (yaooqinn) [inline docs/client/rest/rest_api.md:360]: Delete the specified engine.

BTW, can we accurately delete the specified engine by the following parameters?

File: docs/client/rest/rest_api.md [hunk 4]
```diff
@@ -349,6 +349,43 @@ Gets the local log lines from this batch.
...
+| Name                    | Description                   | Type             |
+|:------------------------|:------------------------------| :--------------- |
+| type                    | the engine type               | String(optional) |
+| sharelevel              | the engine share level        | String(optional) |
+| subdomain               | the engine subdomain          | String(optional) |
+| hive.server2.proxy.user | the proxy user to impersonate | String(optional) |
+
+### GET /admin/engine
+
+Get a list of engines.
```
Comment (yaooqinn) [inline docs/client/rest/rest_api.md:373]: Get a list of satisfied engines.

File: docs/tools/kyuubi-admin.rst [hunk 5]
```diff
@@ -69,3 +69,51 @@ Usage: ``bin/kyuubi-admin refresh config [options] [<configType>]``
      - Description
    * - hadoopConf
      - The hadoop conf used for proxy user verification.
+
+.. _list_engine:
+
+List Engine
```
Comment (yaooqinn) [inline docs/tools/kyuubi-admin.rst:75]: List Engines

File: docs/client/rest/rest_api.md [hunk 6]
```diff
@@ -386,3 +423,15 @@ Gets the local log lines from this batch.
 | totalOperations | How many queries and meta calls                                                                                     | Int       |
 | exception       | The session exception, such as the exception that occur when opening session                                        | Throwable |
 | eventType       | The type of session event                                                                                           | String    |
+
+### Engine
+
+| Name           | Description                                              | Type   |
+| :------------- |:---------------------------------------------------------| :----- |
+| version        | The kyuubi version                                       | String |
```
Comment (yaooqinn) [inline docs/client/rest/rest_api.md:431]: the version of the Kyuubi server that creates this engine instance

File: docs/tools/kyuubi-admin.rst [hunk 7]
```diff
@@ -69,3 +69,51 @@ Usage: ``bin/kyuubi-admin refresh config [options] [<configType>]``
...
+   * - -esl, --engine-share-level
+     - The engine share level. If not specified, it will read the configuration item kyuubi.engine.share.level from kyuubi-defaults.conf.
+   * - -es, --engine-subdomain
+     - The subdomain for the share level of an engine. If not specified, it will read the configuration item kyuubi.engine.share.level.subdomain from kyuubi-defaults.conf.
+   * - --hs2ProxyUser
+     - The proxy user to impersonate. When specified, it will list engines for the hs2ProxyUser.
+
+.. _delete_engine:
+
+Delete Engine
```
Comment (yaooqinn) [inline docs/tools/kyuubi-admin.rst:99]: Delete an Engine

File: docs/client/rest/rest_api.md (hunk 3 above)
Comment (lightning-L) [reply docs/client/rest/rest_api.md:360]: yes, delete operation will delete an engine at a time. When the parameter is not specified, we will read the value from Kyuubi conf. For subdomain, if it is empty, we will use "default". Then we could construct the zk path.

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
thanks @lightning-L , @Yikf @yaooqinn , merging to master

**Pull request closed without merging**